src/network/socket/SocketHandle.hh

```diff
@@ -0,0 +1,37 @@
+<?hh // strict
+
+namespace HH\Lib\Experimental\Network;
+
+use namespace HH\Lib\Experimental\IO;
+
+/**
+ * Contract for a reliable socket-based handle.
+ */
+interface SocketHandle extends Socket, IO\ReadHandle, IO\WriteHandle {
+  /**
+  * Get the address of the remote peer.
+  */
+  public function getRemoteAddress(): string;
+
+  /**
+   * Get the network port used by the remote peer (or NULL if not network port is being used).
+   */
+  public function getRemotePort(): ?int;
+
+  /**
+   * Place the given data in the socket's send queue.
+   * 
+   * Implementations may try an immediate write before placeing data in the send queue.
+   * 
+   * @param string $data Data to be sent.
```

Comment:
Don't add doccomments that don't add any information that isn't already covered by the type and variable/function name